cmd/nse/nse.go

```diff
@@ -0,0 +1,171 @@
+// Copyright (c) 2018 Cisco and/or its affiliates.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"net"
+	"os"
+	"path"
+	"time"
+
+	"path/filepath"
+
+	"github.com/ligato/networkservicemesh/netmesh/model/netmesh"
+	"github.com/ligato/networkservicemesh/pkg/nsm/apis/common"
+	"github.com/ligato/networkservicemesh/pkg/nsm/apis/nsmconnect"
+	"github.com/ligato/networkservicemesh/plugins/nsmserver"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/grpc"
+)
+
+const (
+	// clientConnectionTimeout defines time the client waits for establishing connection with the server
+	clientConnectionTimeout = time.Second * 60
+	// clientConnectionTimeout defines retry interval for establishing connection with the server
+	clientConnectionRetry = time.Second * 2
+)
+
+var (
+	clientSocketPath     = path.Join(nsmserver.SocketBaseDir, nsmserver.ServerSock)
+	clientSocketUserPath = flag.String("nsm-socket", "", "Location of NSM process client access socket")
+	nseSocketName        = flag.String("nse-socket", "nse.ligato.io.sock", "Name of NSE socket whcih will be used by NSM for Connection Request call")
+)
+
+func dial(ctx context.Context, unixSocketPath string) (*grpc.ClientConn, error) {
+	c, err := grpc.DialContext(ctx, unixSocketPath, grpc.WithInsecure(), grpc.WithBlock(),
+		grpc.WithDialer(func(addr string, timeout time.Duration) (net.Conn, error) {
+			return net.DialTimeout("unix", addr, timeout)
+		}),
+	)
+
+	return c, err
+}
+
+func main() {
+	flag.Parse()
+	flag.Set("logtostderr", "true")
```

Comment:
* flag.Set returns an error, we should handle it. 
* Can we rename this to log-to-stderr for readability?